packages/analytics-js-plugins/src/errorReporting/index.ts

```diff
@@ -18,41 +26,66 @@ const ErrorReporting = (): ExtensionPlugin => ({
   deps: [],
   initialize: (state: ApplicationState) => {
     state.plugins.loadedPlugins.value = [...state.plugins.loadedPlugins.value, pluginName];
+    state.reporting.isErrorReportingPluginLoaded.value = true;
+    state.reporting.breadCrumbs.value = [createNewBreadCrumb('Error Reporting Plugin Loaded')];
   },
   errorReporting: {
-    init: (
-      state: ApplicationState,
-      pluginEngine: IPluginEngine,
-      externalSrcLoader: IExternalSrcLoader,
-      logger?: ILogger,
-    ): Nullable<Promise<any>> => {
-      if (!state.source.value?.config || !state.source.value?.id) {
-        return Promise.reject(new Error(INVALID_SOURCE_CONFIG_ERROR));
-      }
-
-      return pluginEngine.invokeSingle(
-        'errorReportingProvider.init',
-        state,
-        externalSrcLoader,
-        logger,
-      );
-    },
     notify: (
-      pluginEngine: IPluginEngine,
-      client: any,
-      error: Error,
+      error: SDKError,
+      errorState: ErrorState,
       state: ApplicationState,
+      httpClient: IHttpClient,
       logger?: ILogger,
     ): void => {
-      pluginEngine.invokeSingle('errorReportingProvider.notify', client, error, state, logger);
+      const { component, tolerateNonErrors, errorFramesToSkip, normalizedError } =
+        getConfigForPayloadCreation(error, errorState.severityReason.type);
+
+      // Generate the error payload
+      const errorPayload = Event.create(
```

Comment:
I don't think we need the help of the Bugsnag library to create the payloads. It's unnecessary to increase the plugin size.
The logic is pretty straightforward. We don't need to handle the exhaustive cases that the "generic" Bugsnag lib needs to.

Let's discuss this offline.
CC @bardisg

Reply:
Addressed